// node-sf-bulk2/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BulkAPI2 = exports.STATE = exports.RESULTTYPE = exports.OPERATION = exports.LINEENDING = exports.COLUMNDELIMITER = exports.CONTENTTYPE = void 0;
const enum_1 = require("./model/enum");
Object.defineProperty(exports, "CONTENTTYPE", { enumerable: true, get: function () { return enum_1.CONTENTTYPE; } });
Object.defineProperty(exports, "COLUMNDELIMITER", { enumerable: true, get: function () { return enum_1.COLUMNDELIMITER; } });
Object.defineProperty(exports, "LINEENDING", { enumerable: true, get: function () { return enum_1.LINEENDING; } });
Object.defineProperty(exports, "OPERATION", { enumerable: true, get: function () { return enum_1.OPERATION; } });
Object.defineProperty(exports, "STATE", { enumerable: true, get: function () { return enum_1.STATE; } });
Object.defineProperty(exports, "RESULTTYPE", { enumerable: true, get: function () { return enum_1.RESULTTYPE; } });
var bulk2_1 = require("./bulk2");
Object.defineProperty(exports, "BulkAPI2", { enumerable: true, get: function () { return __importDefault(bulk2_1).default; } });
