// node-sf-bulk2/dist/model/enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RESULTTYPE = exports.STATE = exports.OPERATION = exports.LINEENDING = exports.COLUMNDELIMITER = exports.CONTENTTYPE = void 0;
var CONTENTTYPE;
(function (CONTENTTYPE) {
    CONTENTTYPE[CONTENTTYPE["CSV"] = 0] = "CSV";
})(CONTENTTYPE = exports.CONTENTTYPE || (exports.CONTENTTYPE = {}));
var COLUMNDELIMITER;
(function (COLUMNDELIMITER) {
    COLUMNDELIMITER[COLUMNDELIMITER["BACKQUOTE"] = 0] = "BACKQUOTE";
    COLUMNDELIMITER[COLUMNDELIMITER["CARET"] = 1] = "CARET";
    COLUMNDELIMITER[COLUMNDELIMITER["COMMA"] = 2] = "COMMA";
    COLUMNDELIMITER[COLUMNDELIMITER["PIPE"] = 3] = "PIPE";
    COLUMNDELIMITER[COLUMNDELIMITER["SEMICOLON"] = 4] = "SEMICOLON";
    COLUMNDELIMITER[COLUMNDELIMITER["TAB"] = 5] = "TAB";
})(COLUMNDELIMITER = exports.COLUMNDELIMITER || (exports.COLUMNDELIMITER = {}));
var LINEENDING;
(function (LINEENDING) {
    LINEENDING[LINEENDING["LF"] = 0] = "LF";
    LINEENDING[LINEENDING["CRLF"] = 1] = "CRLF";
})(LINEENDING = exports.LINEENDING || (exports.LINEENDING = {}));
var OPERATION;
(function (OPERATION) {
    OPERATION[OPERATION["insert"] = 0] = "insert";
    OPERATION[OPERATION["delete"] = 1] = "delete";
    OPERATION[OPERATION["hardDelete"] = 2] = "hardDelete";
    OPERATION[OPERATION["update"] = 3] = "update";
    OPERATION[OPERATION["upsert"] = 4] = "upsert";
})(OPERATION = exports.OPERATION || (exports.OPERATION = {}));
var STATE;
(function (STATE) {
    STATE[STATE["Open"] = 0] = "Open";
    STATE[STATE["UploadComplete"] = 1] = "UploadComplete";
    STATE[STATE["Aborted"] = 2] = "Aborted";
    STATE[STATE["JobComplete"] = 3] = "JobComplete";
    STATE[STATE["Failed"] = 4] = "Failed";
})(STATE = exports.STATE || (exports.STATE = {}));
var RESULTTYPE;
(function (RESULTTYPE) {
    RESULTTYPE[RESULTTYPE["successfulResults"] = 0] = "successfulResults";
    RESULTTYPE[RESULTTYPE["failedResults"] = 1] = "failedResults";
    RESULTTYPE[RESULTTYPE["unprocessedrecords"] = 2] = "unprocessedrecords";
})(RESULTTYPE = exports.RESULTTYPE || (exports.RESULTTYPE = {}));
